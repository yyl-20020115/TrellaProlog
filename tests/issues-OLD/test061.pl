:- initialization(main).

:- use_module(library(lists)).
:- use_module(library(apply)).

main :-
    maplist(col([[1,2],[3,4]]), [1,2], X1),
    writeq(X1), nl,
    maplist(col([[A,2],[3,A]]), [1,2], X2),
    write(X2), nl,
    maplist(length, M, [2,2]),
    writeq(M), nl,
    maplist(col(M), [1,2], X),
    writeq(X), nl.

col(Matrix, N, Column) :-
    maplist(nth1(N), Matrix, Column).
