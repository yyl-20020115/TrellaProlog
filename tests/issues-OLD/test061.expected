[[1,3],[2,4]]
[[_4,3],[2,_4]]
[[_170,_174],[_191,_195]]
[[_170,_191],[_174,_195]]
